### ISSUE TITLE
Improvements to the TorchEnvironmentProvider

### ISSUE THREAD

**Issue by Dankomaister** [atomistic-machine-learning/schnetpack#234]
I noticed that the TorchANI project has committed some changes to the `neighbor_pairs` function used in the `TorchEnvironmentProvider` which should improve performance.

Not sure how much improvements it gives but maybe it could be an idea to implement them.

**Comment by mgastegger** [comment]
I ran some tests, the changes to the `neighbor_pairs` function seem to improve performance on the GPU by a factor of ~3-4 (100 atoms system, 5 Angstrom cutoff). I updated the function in the master branch.

**Comment by Dankomaister** [comment]
Hi @mgastegger 

I finally got a chance to test the updates to the TorchEnvironmentProvider.
From my tests (supercells of TiC) I saw a maximum performance increase of ~18% both on the GPU and CPU (see attached figure). However, I did notice that the updated version uses more memory compared to the old one. With the old version I was able to run a maximum system size of 2000 atoms on my 8GB GTX 1070 (this is for neighbor list calculation only no NN). But with the new version the maximum system size I can run (before running out of memory) is 1458 atoms (with a 5.0 Å cutoff as before).

Also there was an update to the old version that fixed some warnings given with PyTorch 1.6
`UserWarning: This overload of nonzero is deprecated:
        nonzero()
Consider using one of the following signatures instead:
        nonzero(*, bool as_tuple) (Triggered internally at  /opt/conda/conda-bld/pytorch_1595629395347/work/torch/csrc/utils/python_arg_parser.cpp:766.)
  in_cutoff = (distances < cutoff).nonzero()`

So I suggest changing the `in_cutoff = (distances < cutoff).nonzero()` to `in_cutoff = torch.nonzero(distances < cutoff, as_tuple=False)`

![Torch_update](https://user-images.githubusercontent.com/14222532/94771734-e68e8180-03f2-11eb-979e-d14148affe94.png)
